driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -59,6 +60,38 @@ public void setUp() {
         collection.insertOne(Document.parse("{ _id : 0 }"));
     }
 
+    class ChangeStreamWatcher implements Runnable {
+        private volatile boolean interruptedExceptionOccurred = false;
+        private final MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor;
+
+        ChangeStreamWatcher(final MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor) {
+            this.cursor = cursor;
+        }
+
+        @Override
+        public void run() {
+            try {
+                cursor.next();
+            } catch (final MongoInterruptedException e) {
+                interruptedExceptionOccurred = true;
+            } finally {
+                cursor.close();
+            }
+            assertTrue(interruptedExceptionOccurred);
```

Comment:
An assert in a background thread won't cause the test to fail.  This has to move to the test method.

Reply:
Moved the assert to the test and added a getter to the ChangeStreamWatcher class to get the exception boolean value.

Reply:
This assert can now be removed